1Password/onepassword-sdk-python
# Feature: allow forked PRs to access repo secrets after approval

## Summary

This PR enables integration tests to be run on forked PRs, via the `ok-to-test` command.

Credits to https://github.com/imjohnbo/ok-to-test for the idea and the GitHub actions implementation!

## Problem
According to GitHub's official documentation, workflows for PRs against forked repos using `pull_request` events do not have access to the base repository secrets. https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow

As a result, you will not be able to access the secrets in the workflow for PR against forked repository: 


## Thought process
The flow is also explained in this README as well: https://github.com/imjohnbo/ok-to-test

- `integration-test-trusted` is only run on internal PRs and is skipped on forked PRs
- `integration-test-fork` will always be skipped on internal PR and be marked as skipped on forked PRs before the `/ok-to-test sha=commit_sha_to_test` command is added by a collaborator with write permissions to the repo.
- once the `/ok-to-test` command is commented in the PR, the `integration-test-fork` will run on the sdk repo with secrets access and if it's successful it'll update the check pipeline of the forked PR.

## Additional Considerations
A new dependency was added, namely the `peter-evans/slash-command-dispatch@v3` ([very popular repo with 550 stars](https://github.com/peter-evans/slash-command-dispatch)) for using the `/ok-to-test` command which requires a `PERSONAL_ACCESS_TOKEN`. 
We'll be using the 1password-bot for the personal access token: @1PasswordPackagesBot

## How to Test
I tested this flow out in a personal repo. See this PR for how it looks like: https://github.com/AndyTitu/test-ok-to-test/pull/16
